SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -263,24 +278,16 @@ void FSpatialNetGUIDCache::RemoveEntityNetGUID(Worker_EntityId EntityId)
 	// Remove actor subobjects.
 	USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(Driver);
 
-	AActor* Actor = SpatialNetDriver->EntityRegistry->GetActorFromEntityId(EntityId);
-	if (Actor == nullptr)
-	{
-		UE_LOG(LogSpatialPackageMap, Warning, TEXT("Trying to clean up Actor for EntityId %lld but Actor does not exist! Will not cleanup subobjects for this Entity"), EntityId);
-		return;
-	}
-
-	UClass* Class = Actor->GetClass();
-	const FClassInfo& Info = SpatialNetDriver->ClassInfoManager->GetOrCreateClassInfoByClass(Class);
-
 	improbable::UnrealMetadata* UnrealMetadata = SpatialNetDriver->StaticComponentView->GetComponentData<improbable::UnrealMetadata>(EntityId);
 
 	// There are times when the Editor is quitting out of PIE that this is nullptr.
-	if (UnrealMetadata == nullptr)
+	if (UnrealMetadata == nullptr || (IsInGameThread() && IsGarbageCollecting()))
```

Comment:
```suggestion
	if (UnrealMetadata == nullptr || (IsGarbageCollectingOnGameThread()))
```

Reply:
Why was this check added? How was this breaking?

Reply:
I'm forgetting why this was added but I was getting crashes on editor shut down or something similar.

Reply:
Could you remove it to double check?

Reply:
Yep. I have locally, need to push.

Reply:
Have checked and it occurs. Added comment explaining why.
@m-samiec that function is defined in a .cpp and not accessible.